php/post-types/class-block-post.php

```diff
@@ -87,19 +98,25 @@ public function register_controls() {
 		);
 
 		if ( block_lab()->is_pro() ) {
-			$controls = array_merge(
-				$controls,
-				array(
-					'post'     => new Controls\Post(),
-					'taxonomy' => new Controls\Taxonomy(),
-					'user'     => new Controls\User(),
-				)
-			);
+			foreach ( $this->pro_controls as $pro_control ) {
+				$controls[ $pro_control ] = $this->instantiate_pro_control( $pro_control );
```

Comment:
Yeah, that's an interesting idea. 

It would make the codebase a little cleaner. And it wouldn't add much overhead. 

But I thought that because most plugin users don't have the pro license, it'd be best to not instantiate the pro controls when they're not going to have access to them. 

`instantiate_pro_control()` is used in a few places to instantiate a pro control just-in-time.

Reply:
Still, this would make the codebase cleaner, a pretty strong consideration. My opinion above isn't strongly-held.

Reply:
Great thinking. I think the part I don't like is the part where we have to pass the class name as an argument (e.g. `instantiate_control( 'Taxonomy' )`. Feels like a hack to me, especially with the `ucwords`.

What if we:

- Change the method name to `instantiate_control()` since it's not really pro specific anyway
- Even simpler, `get_control()`?
- Check to see that the class _can_ be instantiated before we instantiate it?
- Check to see if the class _is already_ instantiated before we instantiate it?

I'm not sure how all that would fit together – do you think there's some improvements we can pull out from there?

Reply:
Sure, I'm working on this now.

Reply:
@lukecarbis, how does https://github.com/getblocklab/block-lab/pull/306/commits/addc1afeefcb5c777638bd484b5a1e476faa0242#diff-032d8fb82d07a7fc45dc4fd064a36ca8R3 work? It'll require some regression testing to make sure the URL control still works, as this renames the class from `URL` to `Url`.